src/asciidoc/reference/packages/withkojiDispatch.adoc

```diff
@@ -51,19 +48,177 @@ const dispatch = new Dispatch({
+=== .shardName
+
+_String_, name of the shard the client is currently connected to.
+
+[.hcode, id=".userInfo", reftext="userInfo"]
+=== .userInfo
+
+_Object_, the user info of the currently connected client.
+
+== Dispatch methods
+
+[.hcode, id=".connect", reftext="connect"]
+=== .connect()
+
+(Async) Connects the dispatch client to an open shard.
+
+==== Example
+
+[source,javascript]
+dispatch.connect();
 
-Checks the latency of dispatch messages.
+[.hcode, id=".disconnect", reftext="disconnect"]
+=== .disconnect()
+
+(Async) Disconnects the dispatch client from the active connection. 
+
+==== Example
+
+[source,javascript]
+dispatch.disconnect();
+
+[.hcode, id=".emitEvent", reftext="emitEvent"]
+=== .emitEvent(eventName, payload)
+
+Emits the named event to all clients on the current shard.
+
+==== Parameters
+
+* `eventName` – _String_, name of the event.
+* `payload` – _Any_, data to send with event.
+
+==== Example
+
+[source,javascript]
+dispatch.emitEvent('myEvent', myDataPayload);
+
+[.hcode, id=".info", reftext="info"]
+=== .info()
+
+Gets info on the shard the client is currently connected to.
 
 ==== Returns
 
-Number, the latency of the last message in milliseconds.
+(Async) _ShardInfo_, Object describing the shard the client is currently connected to with shard name and number of connected clients.
+
+==== Example
+
+[source,javascript]
+----
+dispatch.info.then((info) => {
+    currentInfo = info;
+});
+----
+
+[.hcode, id=".on", reftext="on"]
+=== .on(eventName, handler)
+
+Sets a listener to a specific event and runs the handler when the event is dispatched over the shard.
+
+TIP: For Koji custom events see <<#_dispatch_event, DISPATCH_EVENT>>.
+
+==== Parameters
+
+* `eventName` – _String_, name of the event to subscribe to.
+* `handler` – _Function_, the handler to run when the event is fired.
+
+==== Example
+
+[source,javascript]
+dispatch.on('myEvent', myHandlerFunction);
+
+[.hcode, id=".removeEventListener", reftext="removeEventListener"]
+=== .removeEventListener(eventName)
+
+Removes listeners from the specified event.
+
+==== Parameters
+
+* `eventName` – _String_, name of the event to remove listeners from.
 
 ==== Example
 
 [source,javascript]
-const myLatency = dispatch.latency();
+dispatch.remveEventListener('myEvent');
+
+[.hcode, id=".setUserInfo", reftext="setUserInfo"]
+=== .setUserInfo(userInfo)
+
+Sets the current client's user info for the currently connected shard.
+
+==== Parameters
+
+* `userInfo` – _Any_, the data for user info to set.
+
+==== Example
+
+[source,javascript]
+dispatch.setUserInfo({username:"myUsername"});
+
+== Dispatch Constants
+=== DISPATCH_EVENT
+
+Constant holding special event keys for Koji Dispatch.
+To subscribe to these events see <<#.on>>.
+
+. `DISPATCH_EVENT.CONNECTED` - Fired when the current client has successfully connected to a shard.
+. `DISPATCH_EVENT.CONNECTED_CLIENTS_CHANGED` - Fired when the list of clients currently connected to the shard changes.
```

Comment:
Also, let's payloads for the events in the descriptions (I missed it in the examples, at first). Similar to our callback arguments, I think.